lib/osf-components/addon/components/addons-service/manager/component.ts

```diff
@@ -85,10 +87,25 @@ export default class AddonsServiceManagerComponent extends Component<Args> {
     }
 
     @action
-    acceptTerms() {
+    async acceptTerms() {
+        await taskFor(this.selectedProvider!.getAuthorizedStorageAccounts).perform();
+        if(this.selectedProvider!.authorizedStorageAccounts!.length > 0){
+            this.pageMode = PageMode.NEW_OR_EXISTING_ACCOUNT;
+        } else {
+            this.pageMode = PageMode.ACCOUNT_CREATE;
+        }
+    }
```

Comment:
I'm a little reluctant about this `async` action here, especially if the `getAuthorizedStorageAccounts` task takes a while or errors. Could it be possible to do something where we route users to `NEW_OR_EXISTING_ACCOUNT` not matter what and handle the loading state of authorized accounts on the template? Something like:

```suggestion
    acceptTerms() {
        taskFor(this.selectedProvider!.getAuthorizedStorageAccounts).perform();
        this.pageMode = PageMode.NEW_OR_EXISTING_ACCOUNT;
    }
```

and then update the template to do something like
```
{{else if (eq manager.pageMode 'newOrExistingAccount')}}
   {{#if this.selectedProvider.getAuthorizedStorageAccounts.isRunning}}
       // some loading state
   {{else}}
       {{#if this.selectedProvider.authorizedStorageAccounts.length}}
           <Button
               data-test-addon-existing-account-button
               data-analytics-name='Choose existing account'
               {{on 'click' manager.chooseExistingAccount}}
           >
               {{t 'addons.accountSelect.existing-account'}}
           </Button>
         {{else}}
            // No accounts found
         {{/if}}
    {{/if}}
    // Create new button below should be the same as before
```

I'm not in love with this idea, as it's another extra click for users that don't have an authorized account and that's kind of annoying.